Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -121,44 +121,69 @@ public typealias OnWebSocketHandshakeInterceptComplete = (HTTPRequestBase, Int32
 /// such as signing/authorization etc... Returning from this function does not continue the websocket
 /// handshake since some work flows may be asynchronous. To accommodate that, onComplete must be invoked upon
 /// completion of the signing process.
-public typealias OnWebSocketHandshakeIntercept = (HTTPRequest, @escaping OnWebSocketHandshakeInterceptComplete) -> Void
+public typealias OnWebSocketHandshakeIntercept = (HTTPRequest, @escaping OnWebSocketHandshakeInterceptComplete) async -> Void
 
 // MARK: - Mqtt5 Client
 public class Mqtt5Client {
     private var rawValue: UnsafeMutablePointer<aws_mqtt5_client>?
-    private var callbackCore: MqttCallbackCore
 
+    ///////////////////////////////////////
+    // user callbacks
+    ///////////////////////////////////////
+    internal let onPublishReceivedCallback: OnPublishReceived
+    internal let onLifecycleEventStoppedCallback: OnLifecycleEventStopped
+    internal let onLifecycleEventAttemptingConnect: OnLifecycleEventAttemptingConnect
+    internal let onLifecycleEventConnectionSuccess: OnLifecycleEventConnectionSuccess
+    internal let onLifecycleEventConnectionFailure: OnLifecycleEventConnectionFailure
+    internal let onLifecycleEventDisconnection: OnLifecycleEventDisconnection
+    // The websocket interceptor could be nil if the websocket is not in use
+    internal let onWebsocketInterceptor: OnWebSocketHandshakeIntercept?
+    internal let rwlock = ReadWriteLock()
+    internal var callbackFlag = true
+
+    /// Creates a Mqtt5Client instance using the provided Mqtt5ClientOptions. Once the Mqtt5Client is created,
```

Comment:
Trivial: "Creates a Mqtt5Client instance using the provided Mqtt5ClientOptions." We can probably remove the first sentence about creation as it's just how an init() is expected to work.

"Once the Mqtt5Client is created, changing the settings will not cause a change in already created Mqtt5Client's." We can also remove this sentence as all options in Mqtt5ClientOptions are const. 

"Once created, it is MANDATORY to call `close()` to clean up the Mqtt5Client resource" Remove "Once created, " and just state that it is mandatory to call `close()`